deltacat/aws/s3u.py

```diff
@@ -227,8 +232,12 @@ def read_file(
     except ClientError as e:
         if e.response["Error"]["Code"] in TIMEOUT_ERROR_CODES:
             # Timeout error not caught by botocore
-            raise RetryableError(f"Retry table download from: {s3_url}") from e
-        raise NonRetryableError(f"Failed table download from: {s3_url}") from e
+            raise RetryableDownloadTableError(
+                msg=f"Retry table download from: {s3_url}", s3_url=s3_url
+            ) from e
+        raise DownloadTableError(
+            msg=f"Failed table download from: {s3_url}", s3_url=s3_url
+        ) from e
     except BaseException as e:
```

Comment:
Should we convert this unknown exception into one of the known exceptions in `deltacat.exceptions`? In the current state, we will just see our reliability go down.

Reply:
This is not resolved. Are you planning on resolving it?

Reply:
Are you suggesting we do not change the original implementation, just throw unretryable error instead?

Reply:
All known errors must be handled here, otherwise our reliability will go down. You may explore dynamically handling these via a scaling param as well.